sql/hive/src/main/scala/org/apache/spark/sql/hive/hiveUdfs.scala

```diff
@@ -81,6 +82,8 @@ private[hive] case class HiveSimpleUdf(funcWrapper: HiveFunctionWrapper, childre
 
   override def nullable: Boolean = true
 
+  override def checkInputDataTypes(): TypeCheckResult = TypeCheckResult.TypeCheckSuccess
```

Comment:
i think we do not need to check data types for hive udf. because that be done in   protected lazy val method =
    function.getResolver.getEvalMethod(children.map(_.dataType.toTypeInfo))
if data types are mismatched, this function will throw NoMatchingMethodException.